ssl/quic/quic_rx_depack.c

```diff
@@ -0,0 +1,612 @@
+                int stream_type = ssl_get_stream_type(stream);
+
+                ssl_close_stream(stream);
+                if ((stream_type & SSL_STREAM_TYPE_W) != 0) {
+                    OSSL_STATM *statm = GET_CONN_STATM(connection);
+
+                    ssl_close_connection(connection);
+                    ossl_statm_set_error_state(statm, QUIC_STREAM_STATE_ERROR);
+
+                    /*
+                     * There's no point to continue with this packet.  However,
+                     * processing it was a success, so we return with a happy
+                     * face, and leave it to the state machine what to do with
+                     * the general STREAM_STATE_ERROR state.
+                     */
+                    return 1;
+                }
+                break;
+            }
+            }
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_STOP_SENDING: {
+            OSSL_QUIC_FRAME_STOP_SENDING frame_data;
+
+            if (!ossl_quic_wire_decode_frame_stop_sending(pkt, &frame_data))
+                return 0;
+
+            switch (mode) {
+            case collect_ack_data:
+                /* This frame makes the packet ACK eliciting */
+                ackm_data->is_ack_eliciting = 1;
+                break;
+            case process_frames: {
+                QUIC_STREAM *stream
+                    = ssl_get_stream(connection, frame_data.stream_id);
+                int stream_type = ssl_get_stream_type(stream);
+
+                ssl_close_stream(stream);
+                if ((stream_type & SSL_STREAM_TYPE_R) != 0) {
+                    OSSL_STATM *statm = GET_CONN_STATM(connection);
+
+                    ssl_close_connection(connection);
+                    ossl_statm_set_error_state(statm, QUIC_STREAM_STATE_ERROR);
+
+                    /*
+                     * There's no point to continue with this packet.  However,
+                     * processing it was a success, so we return with a happy
+                     * face, and leave it to the state machine what to do with
+                     * the general STREAM_STATE_ERROR state.
+                     */
+                    return 1;
+                }
+                break;
+            }
+            }
+            break;
+        }
+
+        case OSSL_QUIC_FRAME_TYPE_CRYPTO: {
+            OSSL_QUIC_FRAME_CRYPTO frame_data;
+
+            if (!ossl_quic_wire_decode_frame_crypto(pkt, &frame_data))
+                return 0;
+
+            switch (mode) {
+            case collect_ack_data:
+                /* This frame makes the packet ACK eliciting */
+                ackm_data->is_ack_eliciting = 1;
+                break;
+            case process_frames: {
+                /* ADD CODE to send |frame_data.data| to the handshake manager */
+                break;
+            }
+            }
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_NEW_TOKEN: {
```

Comment:
Some `case` statements seem to have a blank line before them, and others don't

Reply:
Yes.  I did that with cases that are treated as one and the same, for clearer grouping...  I didn't do that consistently, though.